### PR TITLE
Aaa/apis08d view authorized application

### LINKED ISSUE #30542: APIS08d - Authorized Application - View detail of an Authorized Application
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

This is a new feature. It aims to make the Authorized Apps manageable in the BO

### Describe the solution you'd like

As a shop administrator
I want to be able to view the detail of an authorized application in the Authz server management page in the BO
In order to view the description of an authorized app and its API Accesses

### Alternatives you've considered

_No response_

### Additional context

[Figma link](https://cutt.ly/2MUXEgy)

### Acceptance test
AT1

1. Given that
   1. I am a shop administrator
   2. The authorization server administration is available
2. When I open the the authz server administration page in the BO
3. Then
   1. I can see the list of applications that have an API access to my shop
   2. I can click on an Authorized App to view its details

### QA Scenarios
* Functional > BO > Advanced Parameters > Authorization Server > View detail

|**Action** | **Additional** | **Expected** |
|-----------|-----------|-----------|
| | **PRE-COND : Enable experimental feature : Authorization server** | |
In BO, Go to Advanced Parameters > New & Experimental Features |   | New & Experimental Features Page is displayed correctly.
Set checked the checkbox “Authorization server” |   | The checkbox “Authorization server” is check
Click on Save button |   | A modal appeared
Click on Enable button in the modal | | The message “Update successful” is displayed
| | **PRE-COND : Create an application** | |
In BO, Go to Advanced Parameters > Authorization Server |   | Authorization Server Page is displayed correctly.<br>No records found in the table
Click on the button "Add new authorized app"|   | New authorized application Page is displayed correctly
Fill the form with following data (App Name : "Application XYZ", Description: "Description ABC")<br>Click on Save button| | The message “Successful creation” is displayed<br>There is 1 application in the list
| | **Main Test** | |
In BO, Go to Advanced Parameters > Authorization Server |   | Authorization Server Page is displayed correctly.<br>1 record found in the table
On the first row, click on the row | | - Application information Page is displayed correctly<br>- Application name & Description are displayed.<br>- No record founds for API access
| | **POSTCOND : Remove an application** | |
In BO, Go to Advanced Parameters > Authorization Server |   | Authorization Server Page is displayed correctly.<br>1 record found in the table
On the first row, click on the button “Three points” | | The dropdown is displayed.
Click on the Delete Button in the dropdown | | A modal appeared
Click on the Confirm Button in the modal | | The message “Successful deletion” is displayed<br>No records found in the table
| | **POSTCOND : Disable experimental feature : Authorization server** | |
In BO, Go to Advanced Parameters > New & Experimental Features |   | New & Experimental Features Page is displayed correctly.
Set unchecked the checkbox “Authorization server” |   | The checkbox “Authorization server” is unchecked
Click on Save button |   |  The message “Update successful” is displayed

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | View the details of authorized application
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See #30542 
| Fixed ticket?     | Fixes #30542 
| Related PRs       | Waiting for #30938 
| Sponsor company   | https://github.com/PrestaShopCorp

### CODE REVIEW

**Review by tleon — DISMISSED** [review]

File: src/Core/Domain/AuthorizationServer/ValueObject/ApplicationIdInterface.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\ValueObject;
+
+/**
+ * @see ApplicationId
+ */
+interface ApplicationIdInterface
```
Comment (jolelievre) [inline src/Core/Domain/AuthorizationServer/ValueObject/ApplicationIdInterface.php]: Seems like this interface is not used, and we rarely use interfaces on values objects The rare exception are when we have for example:
- `CurrencyId` force a valid int value > 0
- `NoCurrencyId` hard-coded value with 0 that is used to remove an association (different from `null` which means no changes for partial updates)

In these cases the interface can be useful to allow using either one or the other VO

File: src/Core/Grid/Query/ApiAccessQueryBuilder.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Search\Filters\ApiAccessesFilters;
+
+/**
+ * Class ApiAccessQueryBuilder builds search & count queries for api access grid.
+ */
+final class ApiAccessQueryBuilder extends AbstractDoctrineQueryBuilder
```
Comment (jolelievre) [inline src/Core/Grid/Query/ApiAccessQueryBuilder.php]:
```suggestion
class ApiAccessQueryBuilder extends AbstractDoctrineQueryBuilder
```
Not sure having a final is really relevant for us, however allowing extension can be intresting for modules

**Review by 0x346e3730 — DISMISSED** [review]
LGTM 👍

**Review by M0rgan01 — APPROVED** [review]

**Review by boherm — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by djoelleuch** [comment]
Hello @mflasquin  ,
I cannot test this PR because it's blocked by this [issue ](https://github.com/PrestaShop/PrestaShop/issues/32881git )

Thank you !

**Comment by mflasquin** [comment]
Hi @djoelleuch https://github.com/PrestaShop/PrestaShop/issues/32881 is fixed :)

**Review by djoelleuch — DISMISSED** [review]
Hello @mflasquin ,
automated tests : OK

[Untitled_ Jun 21, 2023 4_57 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/115152460/50f473ff-621b-4e0f-8788-416cf0897079)

It's QA approved :heavy_check_mark:
Thank you :rocket:

**Comment by nicosomb** [comment]
@mflasquin you have a conflict

**Review by nicosomb — DISMISSED** [review]

**Review by M0rgan01 — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @mflasquin

**Pull request merged**